src/test/java/hudson/plugins/git/GitSCMTest.java

```diff
@@ -1268,6 +1268,70 @@ public Void invoke(Repository repo, VirtualChannel channel) throws IOException,
             }
         });
     }
+    
+    /**
+     * Verifies that if project specifies LocalBranch with value of "**" 
+     * that the checkout to a local branch using remote branch name sans 'origin'.
+     * This feature is necessary to support Maven release builds that push updated
+     * pom.xml to remote branch as 
+     * <br/>
+     * <pre>
+     * git push origin localbranch:localbranch
+     * </pre>
+     * @throws Exception
+     */
+    public void testCheckoutToDefaultLocalBranch_StarStar() throws Exception {
+       FreeStyleProject project = setupSimpleProject("master");
+
+       final String commitFile1 = "commitFile1";
+       commit(commitFile1, johnDoe, "Commit number 1");
+       GitSCM git = (GitSCM)project.getScm();
+       git.getExtensions().add(new LocalBranch("**"));
+       FreeStyleBuild build1 = build(project, Result.SUCCESS, commitFile1);
+
+       assertEquals("GIT_BRANCH", "origin/master", getEnvVars(project).get(GitSCM.GIT_BRANCH));
+       assertEquals("GIT_LOCAL_BRANCH", "master", getEnvVars(project).get(GitSCM.GIT_LOCAL_BRANCH));
+    }
+
+    /**
+     * Verifies that if project specifies LocalBranch with null value (empty string) 
+     * that the checkout to a local branch using remote branch name sans 'origin'.
+     * This feature is necessary to support Maven release builds that push updated
+     * pom.xml to remote branch as 
+     * <br/>
+     * <pre>
+     * git push origin localbranch:localbranch
+     * </pre>
+     * @throws Exception
+     */
+    public void testCheckoutToDefaultLocalBranch_NULL() throws Exception {
+       FreeStyleProject project = setupSimpleProject("master");
+
+       final String commitFile1 = "commitFile1";
+       commit(commitFile1, johnDoe, "Commit number 1");
+       GitSCM git = (GitSCM)project.getScm();
+       git.getExtensions().add(new LocalBranch(""));
+       FreeStyleBuild build1 = build(project, Result.SUCCESS, commitFile1);
+
+       assertEquals("GIT_BRANCH", "origin/master", getEnvVars(project).get(GitSCM.GIT_BRANCH));
+       assertEquals("GIT_LOCAL_BRANCH", "master", getEnvVars(project).get(GitSCM.GIT_LOCAL_BRANCH));
+    }
+
+    /**
+     * Verifies that GIT_LOCAL_BRANCH is not set if LocalBranch extension
+     * is not configured.
+     * @throws Exception
+     */
+    public void testCheckoutSansLocalBranchExtension() throws Exception {
+       FreeStyleProject project = setupSimpleProject("master");
+
+       final String commitFile1 = "commitFile1";
+       commit(commitFile1, johnDoe, "Commit number 1");
+       FreeStyleBuild build1 = build(project, Result.SUCCESS, commitFile1);
+
+       assertEquals("GIT_BRANCH", "origin/master", getEnvVars(project).get(GitSCM.GIT_BRANCH));
+       assertEquals("GIT_LOCAL_BRANCH", null, getEnvVars(project).get(GitSCM.GIT_LOCAL_BRANCH));
+    }
 
```

Comment:
I suspect the environment variable is not null, instead it is the empty string.  I'll check as part of the evaluation of the pull request.